### PR TITLE
Decentraland: Implement can execute

### PR DESCRIPTION
**Pull request by facuspagnuolo**

### CODE REVIEW

File: packages/decentraland-1/contracts/actions/OTCSwapper.sol [hunk 1]
```diff
@@ -39,13 +48,21 @@ contract OTCSwapper is BaseSwapper {
     }
 
     function _call(address tokenIn, address tokenOut, uint256 amountOut, uint256 slippage) internal {
```
Comment (dmf7z) [inline packages/decentraland-1/contracts/actions/OTCSwapper.sol:50]: I first thought that OTC traders should not depend on Chainlink price so call function should be like this:

`_call(address tokenIn, address tokenOut, uint256 amountIn, uint256 minAmountOut)`

Then you calculate the amountOut using ChainLink + configured slippage. If amountOut >= minAmountOut, do the swap.

File: packages/decentraland-1/contracts/actions/OTCSwapper.sol [hunk 2]
```diff
@@ -39,13 +48,21 @@ contract OTCSwapper is BaseSwapper {
     }
```
[the remark this answers is not in the document]
Comment (facuspagnuolo) [reply packages/decentraland-1/contracts/actions/OTCSwapper.sol:48]: There is no harm on leaving it right? I mean, if we need to do it as relayers we can redeem gas costs too

File: packages/decentraland-1/contracts/actions/OTCSwapper.sol (hunk 1 above)
Comment (facuspagnuolo) [reply packages/decentraland-1/contracts/actions/OTCSwapper.sol:50]: But then you won't know exactly how much you need to approve to the smart vault beforehand

File: packages/decentraland-1/contracts/actions/OTCSwapper.sol (hunk 2 above)
[the remark this answers is not in the document]
Comment (dmf7z) [reply packages/decentraland-1/contracts/actions/OTCSwapper.sol:55]: sure, no harm

**Review by dmf7z — APPROVED** [review]
👍

**Pull request merged**